default/python/AI/ColonisationAI.py

```diff
@@ -154,6 +154,54 @@ def max_pop_size():
     return max_pop_size()
 
 
+def get_empire_detection(empire_id):
+    # TODO: move to an EspionageAI module
+    # TODO doublecheck typical AI research times for Radar, for below default value
+    empire_detection = 10 if fo.currentTurn() < 40 else 30
+    empire = None
+    if empire_id != ALL_EMPIRES:
+        empire = fo.getEmpire(empire_id)
+    if empire:
+        # TODO: expose Empire.GetMeter to automatically take into account detection specials
+        for techname in sorted(AIDependencies.DETECTION_STRENGTHS, reverse=True):
+            if empire.techResearched(techname):
+                empire_detection = AIDependencies.DETECTION_STRENGTHS[techname]
+                break
+    return empire_detection
```

Comment:
Also add TODO to account for stealth-giving buildings

Reply:
done